### PR TITLE
Add new Prometheus alert `ApiserverRequestsFailureRate` for API Server failure rate

### PR DESCRIPTION
**Pull request by cathyzhang05**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Add api server failure rate high alert to control plane

**Special notes for your reviewer**:

**Release note**:

```other operator
Add new Prometheus alert `ApiserverRequestsFailureRate` for API Server failure rate. 
```

### CODE REVIEW

**Comment by shafeeqes** [comment]
Please adapt the unit tests.

File: pkg/operation/botanist/component/kubeapiserver/monitoring.go [hunk 1]
```diff
@@ -184,6 +184,17 @@ const (
...
+  - alert: ApiserverRequestsFailureRate10mins
+    expr: max(sum by(instance,resource,verb) (rate(` + monitoringMetricApiserverRequestTotal + `{code=~"5.."}[10m])) / sum by(instance,resource,verb) (rate(` + monitoringMetricApiserverRequestTotal + `[10m]))) * 100
+    for: 30m
+    labels:
+      severity: warning
+      type: seed
+      visibility: owner
+    annotations:
+      description: The Apiserver requests failure rate exceed 0.3%.
+      summary: Kubernetes API server failure rate is high
```
Comment (istvanballok) [inline pkg/operation/botanist/component/kubeapiserver/monitoring.go:198]: There are several issues here, please review / double check.

- please make sure the fields and texts are consistent (30m vs 10mins)
- is 0.3% high? maybe you meant 30%? Anyways, the comparison with this value is missing in the promql expression. Without such a comparison (`>0.3`) the alert would always fire
- please watch out for typo-s (exceed, exceeds)

please consider using the local setup to test your changes: https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md

**Review by istvanballok — CHANGES_REQUESTED** [review]
Please fix the typo in the title and the description: `plan` -> `plane` and make sure to add a release note.

Also note that the title prefix "landscape issue 66, " contains a prominent reference to an SAP internal repository which is not so meaningful in this public g/g repository. Please check some other PRs in this repository and follow their example by removing the mentioned prefix from the title.

**Comment by rfranzke** [comment]
@cathyzhang05 What is the status of this PR? Do you still plan to continue working on it?

**Comment by cathyzhang05** [comment]
> @cathyzhang05 What is the status of this PR? Do you still plan to continue working on it?

Hi @rfranzke , previously I have some trouble to setup local environment, right now it has been resolved. I am testing locally. And adjusting the threshold values.

**Review by istvanballok — APPROVED** [review]
/lgtm

Please fix the typo (plan vs. plane) in the PR description and add a sentence as a release note.

**Comment by oliver-goetz** [comment]
/retitle Add new Prometheus alert `ApiserverRequestsFailureRate` for API Server failure rate
/approve

**Pull request merged**